bugfix: serial keyword can be specified as a percentage

klshxsh opened the pull request:
The serial keyword can be specified as a percentage, e.g '30%' or a number e.g. '3'; therefore it needs to be a string to allow both types

Commit:
The serial keyword can be specified as a percentage, e.g '30%' or a number e.g. '3'; therefore it needs to be a string to allow both types

jimi-c commented on lib/ansible/playbook/play.py:
```diff
@@ -87,7 +87,7 @@ class Play(Base, Taggable, Become):
     _force_handlers      = FieldAttribute(isa='bool', always_post_validate=True)
     _max_fail_percentage = FieldAttribute(isa='percent', always_post_validate=True)
-    _serial              = FieldAttribute(isa='int', default=0, always_post_validate=True)
+    _serial              = FieldAttribute(isa='string', default='0', always_post_validate=True)
```
This should be `isa='percent'` (just like _max_fail_percentage above it).

klshxsh committed:
changed 'string' to 'percent' for _serial

Note that this allows both integers (e.g. 3) and percentages (e.g. "30%")
Also changed default back to 0 rather than '0'

klshxsh commented:
Thanks - I have made the suggested change (from string to percent)

jimi-c commented:
Hi @klshxsh, this was actually my mistake - because of the way the code is written, changing the value to a `percentage` type broke things, so I fixed it and tweaked a few other things as well. I've also added a unit test for `_get_serialized_batches()` which should hopefully catch any future bugs.

Thanks!

klshxsh commented:
No worries, thanks.
